docs/architecture/overview.md

```diff
@@ -0,0 +1,163 @@
+ - Create the pull request in system-tests
+
+However, there are many scenarios where a test may not be so simple to implement.
+
+This document aims to give a working understanding of the parts of system-tests, and how to troubleshoot them.
+
+## What are the components of a running test?
+
+When the system tests are executing, there are several main containers of concern.
+ - [Tests Container](TODO)
+   - Responsible for running the actual tests, sending traffic, and asserting results
+ - [Application Container](#structure-of-the-application-container)
+   - Swappable webapp language module that must meet an interface
+ - [Application Proxy](TODO)
+   - Mechanism to inspect payloads from the datadog libraries
+ - [Agent](TODO)
+   - Basic Datadog agent image
+ - [Agent Proxy](TODO)
+   - Mechanism to inspect payloads from the Agent to the Backend
+
+```mermaid
+flowchart TD
+    TESTS[Tests Container] -->|Send Requests| APP
+    APP[Application Container] --> APPPROXY
+    APPPROXY[Application Proxy] --> AGENT
+    AGENT[Agent Container] --> AGENTPROXY
+    AGENTPROXY[Agent Proxy] -->|remote request| BACKEND
+    BACKEND[Datadog]
+```
+
+The tests send requests directly to the application.
+The tests then wait on the results, which are available as the logs are collected from [mitmproxy](TODO) dumps.
+
+## What are system-tests bad for?
+
+ - Combinatorial style tests
+ - Tests that require specific versions of runtimes, libraries, or operating systems
+ - Cloud deployments, kubernetes, distributed deployments
+ - Immediately knowing the reason a feature fails
+ - Problems or features which are not shared across tracers
+ - Performance or throughput testing
+
+ *Examples of bad candidates:*
+  - The .NET tracer must not write invalid IL for it's earliest supported runtime
+  - The startup overhead of the Java tracer is less than 3s for a given sample application
+  - The python tracer must not fail to retrieve traces for a version range of the mongodb library
+
+## What are system-tests good for?
+
+ - Catching regressions on shared features
+ - Wide coverage in a short time frame
+ - Shared test coverage across all tracer libraries
+ - Ensuring requirements for shared features are met across tracer libraries
+
+*Examples of good candidates:*
+  - `DD_TAGS` must be parsed correctly and carried as tags on all traces
+  - Tracer libraries must be able to communicate with the agent through Unix Domain Sockets
+  - Sampling rates from the agent are respected when not explicitly configured
+  - All tracer libraries log consistent diagnostic information at startup
+
+## How do I add a new test?
+
+The default folder to add new tests is `./tests`.
+
+The framework used for running tests is [pytest](https://docs.pytest.org/).
+
+For a test to be run, it must have the filename prefix `test_`.
+
+Follow the example provided within `./tests/template_test.py` if you are unfamiliar with pytest.
```

Comment:
i like the idea of a template.
nit: maybe add direct link to `./tests/template_test.py`

Reply:
done